common/src/main/java/com/fasterxml/jackson/databind/introspect/GuiceInjectableValues.java

```diff
@@ -36,6 +37,17 @@ public Object findInjectableValue(
       Object valueId, DeserializationContext ctxt, BeanProperty forProperty, Object beanInstance
   )
   {
-    return injector.getInstance((Key) valueId);
+    if(Key.class.isAssignableFrom(valueId.getClass())){
```

Comment:
This may not be that odd, the documentation for `InjectableValues` states that `valueId` can be an *"Object that identifies value to inject; may be a simple name or more complex identifier object, whatever provider needs"*

It might make sense to add a comment though as to why this check is needed.